x-pack/legacy/plugins/index_management/public/components/mappings_editor/components/document_fields/fields/edit_field/edit_field_header_form.tsx

```diff
@@ -4,54 +4,123 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
+import React, { ChangeEvent } from 'react';
 import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+
 import {
-  TextField,
+  FormDataProvider,
   SelectField,
   UseField,
   FieldConfig,
-  ValidationFunc,
+  FieldHook,
 } from '../../../../shared_imports';
-import { ParameterName } from '../../../../types';
-import { FIELD_TYPES_OPTIONS, PARAMETERS_DEFINITION } from '../../../../constants';
+import { MainType, ParameterName, Field } from '../../../../types';
+import {
+  TYPE_DEFINITION,
+  FIELD_TYPES_OPTIONS,
+  MULTIFIELD_TYPES_OPTIONS,
+  PARAMETERS_DEFINITION,
+} from '../../../../constants';
+import { NameParameter } from '../../field_parameters';
 
 const getFieldConfig = (param: ParameterName): FieldConfig =>
   PARAMETERS_DEFINITION[param].fieldConfig || {};
 
 interface Props {
-  uniqueNameValidator: ValidationFunc;
+  isMultiField: boolean;
+  defaultValue: Field;
 }
 
-export const EditFieldHeaderForm = ({ uniqueNameValidator }: Props) => {
-  const { validations, ...rest } = getFieldConfig('name');
-  const nameConfig: FieldConfig = {
-    ...rest,
-    validations: [
-      ...validations!,
-      {
-        validator: uniqueNameValidator,
-      },
-    ],
+export const EditFieldHeaderForm = ({ isMultiField, defaultValue }: Props) => {
+  const onTypeChange = (field: FieldHook) => (e: ChangeEvent<HTMLSelectElement>) => {
+    const { value } = e.currentTarget;
+    const typeDefinition = TYPE_DEFINITION[value as MainType];
+    const hasSubType = typeDefinition.subTypes !== undefined;
+    const subTypeField = field.form.getFields().subType;
```

Comment:
This would not work as `hasSubType` could be `true` from the `TYPE_DEFINITION` but the `subTypeField` might not exist in the DOM yet at this moment (when switching from a type that does not have a subType to one that does have one). In that case, L102 would render on the next tick

```js

// L102

{hasSubType && (
  <EuiFlexItem grow={false}>
    <UseField
      path="subType"
```